src/Symfony/Bundle/FrameworkBundle/Client.php

```diff
@@ -163,34 +163,36 @@ protected function getScript($request)
     {
         $kernel = str_replace("'", "\\'", serialize($this->kernel));
         $request = str_replace("'", "\\'", serialize($request));
+        $errorReporting = error_reporting();
 
-        $r = new \ReflectionObject($this->kernel);
+        $requires = '';
+        foreach (get_declared_classes() as $class) {
+            if (0 === strpos($class, 'ComposerAutoloaderInit')) {
+                $r = new \ReflectionClass($class);
+                $file = dirname(dirname($r->getFileName())).'/autoload.php';
+                if (file_exists($file)) {
+                    $requires .= "require_once '".str_replace("'", "\\'", $file)."';\n";
+                }
+            }
+        }
 
-        $autoloader = dirname($r->getFileName()).'/autoload.php';
-        if (is_file($autoloader)) {
-            $autoloader = str_replace("'", "\\'", $autoloader);
-        } else {
-            $autoloader = '';
+        if (!$requires) {
+            throw new \RuntimeException('Composer autoloader not found.');
         }
 
-        $path = str_replace("'", "\\'", $r->getFileName());
+        $requires .= "require_once '".str_replace("'", "\\'", (new \ReflectionObject($this->kernel))->getFileName())."';\n";
```

Comment:
Dunno :) Currently, this is consistent with other places (HttpKernel's Client) and with the previous code. I don't see the need to diverge here.